File: Programmers/오픈 채팅방.py
def solution(record):
    answer = []
    user = {}
    for key in record:
        temp = key.split(" ")
        if temp[0] == "Enter" or temp[0] == "Change":
            user[temp[1]] = temp[2]

    for key in record:
        temp = key.split(" ")
        if temp[0] == "Enter":
            answer.append(user[temp[1]] + "님이 들어왔습니다.")
        elif temp[0] == "Leave":
            answer.append(user[temp[1]] + "님이 나갔습니다.")
    return answer

File: Programmers JS/오픈채팅방.js
function solution(record) {
  var answer = [];
  let user = {};
  record.forEach((log) => {
    log = log.split(" ");
    if (log[0] !== "Leave") {
      user[log[1]] = log[2];
    }
  });
  record.forEach((log) => {
    log = log.split(" ");
    if (log[0] === "Enter") {
      answer.push(user[log[1]] + "님이 들어왔습니다.");
    } else if (log[0] === "Leave") {
      answer.push(user[log[1]] + "님이 나갔습니다.");
    }
  });
  return answer;
}
